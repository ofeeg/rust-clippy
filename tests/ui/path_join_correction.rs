#![allow(unused)]
#![warn(clippy::path_join_correction)]

fn main() {
  // should be linted
  let path = std::path::Path::new("/bin");
  path.join("/sh");
  println!("{}", path.display());

  //should not be linted
  let path = std::path::Path::new("/bin");
  path.join("sh");
  println!("{}", path.display());
  
}
