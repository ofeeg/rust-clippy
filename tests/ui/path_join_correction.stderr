error: argument in join called on path contains a starting '/'
  --> $DIR/path_join_correction.rs:8:10
   |
LL |     path.join("/sh");
   |          ^^^^^^^^^^^ help: try removing first '/': `join("your/path/here")`
   |
   = note: `-D clippy::path-join-correction` implied by `-D warnings`

error: aborting due to previous error

